sympy/testing/hypothesis.py

```diff
@@ -0,0 +1,76 @@
+from hypothesis import strategies as st
+from hypothesis import assume
+from sympy.abc import x
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.polytools import Poly, lcm, gcd
+from typing import Any
+
+
+@st.composite
+def coefficients(draw: st.DrawFn, empty=True, degree=None, rational=False):
+    min_size = 0 if empty else 1
+    elements_strat = Any
+    group = Any
+    if rational:
+        group = QQ
+        elements_strat = st.fractions()
+    else:
+        group = ZZ
+        elements_strat = st.integers()
+    # converting to the underlying type
+    raw_l = draw(st.lists(elements_strat, min_size=min_size, max_size=degree))
+    l = [group(i) for i in raw_l]
+    # ensuring there is no leading zero or fully zero list.
+    if len(l) > 0:
+        assume(any(l))
+        assume(l[0] != 0)
+    return l
+
+
+@st.composite
+def polys(
+    draw: st.DrawFn, empty=True, sparse=False, domain="ZZ", gens=(x,), degree=None
+):
+    if sparse:
+        raise NotImplementedError("Sparse polynomials are not yet supported.")
+    else:
+        return Poly(draw(coefficients(empty, degree=degree)), *gens, domain=domain)
+
+
+def lattice_axioms_singular(polysList, func=lcm):
+    if not polysList:
+        return ValueError("Empty list of polynomials.")
+    f, g, h = polysList[0], polysList[1], polysList[2]
```

Comment:
You can require three polys as positional arguments, and the func as keyword argument:
```suggestion
def lattice_axioms_singular(f, g, h, /, *, func=lcm):
```
(positional-only arguments were new in Python 3.8, which is the oldest version Sympy supports)

Reply:
What is the `/, *` for?